tests/processor/document/test_candidate_relation_adder.py

```diff
@@ -0,0 +1,347 @@
+ENTITY_TEXT_01_BERLIN = LabeledSpan(start=14, end=20, label="city")
+ENTITY_TEXT_01_KARL = LabeledSpan(start=48, end=52, label="person")
+TEXT_01_SENTENCE1 = LabeledSpan(start=0, end=21, label="sentence")
+REL_JANE_LIVES_IN_BERLIN = BinaryRelation(
+    head=ENTITY_TEXT_01_JANE, tail=ENTITY_TEXT_01_BERLIN, label="lives_in"
+)
+
+ENTITY_TEXT_02_SEATTLE = LabeledSpan(start=0, end=7, label="city")
+ENTITY_TEXT_02_JENNY = LabeledSpan(start=25, end=37, label="person")
+TEXT_02_SENTENCE1 = LabeledSpan(start=0, end=24, label="sentence")
+TEXT_02_SENTENCE2 = LabeledSpan(start=25, end=58, label="sentence")
+REL_JENNY_MAYOR_OF_SEATTLE = BinaryRelation(
+    head=ENTITY_TEXT_02_JENNY, tail=ENTITY_TEXT_02_SEATTLE, label="mayor_of"
+)
+
+ENTITY_TEXT_03_KARL = LabeledSpan(start=0, end=4, label="person")
+ENTITY_TEXT_03_BERLIN = LabeledSpan(start=26, end=32, label="city")
+TEXT_03_SENTENCE1 = LabeledSpan(start=0, end=33, label="sentence")
+
+TEXT_01_TOKENS = [
+    "[CLS]",
+    "Jane",
+    "lives",
+    "in",
+    "Berlin",
+    ".",
+    "this",
+    "is",
+    "no",
+    "sentence",
+    "about",
+    "Karl",
+    "[SEP]",
+]
+TEXT_02_TOKENS = [
+    "[CLS]",
+    "Seattle",
+    "is",
+    "a",
+    "rainy",
+    "city",
+    ".",
+    "Jenny",
+    "Du",
+    "##rka",
+    "##n",
+    "is",
+    "the",
+    "city",
+    "'",
+    "s",
+    "mayor",
+    ".",
+    "[SEP]",
+]
+TEXT_03_TOKENS = ["[CLS]", "Karl", "enjoys", "sunny", "days", "in", "Berlin", ".", "[SEP]"]
+
+
+@pytest.fixture
+def candidate_relation_adder():
+    return CandidateRelationAdder(
+        label="no_relation",
+        collect_statistics=True,
+    )
+
+
+@pytest.fixture
+def candidate_relation_adder_with_statistics():
+    return CandidateRelationAdder(
+        label="no_relation",
+        collect_statistics=True,
+        max_distance=8,
+    )
+
+
+@pytest.fixture
+def candidate_relation_adder_without_sort_by_distance():
+    return CandidateRelationAdder(
+        label="no_relation",
+        sort_by_distance=False,
+    )
+
+
+@pytest.fixture
+def candidate_relation_adder_with_no_relation_upper_bound():
+    return CandidateRelationAdder(
+        label="no_relation",
+        added_relations_upper_bound_factor=3,
+    )
+
+
+@pytest.fixture
+def candidate_relation_adder_with_partition():
+    return CandidateRelationAdder(
+        label="no_relation",
+        use_partition=True,
+    )
+
+
+@pytest.fixture
+def candidate_relation_adder_with_partition_and_max_distance():
+    return CandidateRelationAdder(
+        label="no_relation",
+        use_partition=True,
+        max_distance=8,
+    )
+
+
+def test_candidate_relation_adder(candidate_relation_adder):
+    @dataclasses.dataclass
+    class TestDocument(TextDocument):
+        partition: AnnotationList[Span] = annotation_field(target="text")
+        entities: AnnotationList[LabeledSpan] = annotation_field(target="text")
+        relations: AnnotationList[BinaryRelation] = annotation_field(target="entities")
```

Comment:
Is this the same for all tests? If so, just define it once outside the test methods.